### DISCUSSION TITLE
.

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uAssets#20483] [General]

**Comment by ryanbr** [comment]
https://github.com/brave/uBlock/commit/027c7a4fb5b74207b7dedcbc83f4c70527ea5566

Should only affect ios currently. Recent quote issues, causing issues. We are aware of it, and fix is being implemented.

Best way to excempt the filters for the time being. https://github.com/brave/adblock-lists/commit/80c536bc18367e67f8978b37d8876d17c726a775

**Comment by MasterKia** [comment]
What you're essentially asking is for volunteers to accomadate to a browser backed by a company while EasyList still doesn't consider using uBO's extended syntax.

**Reply by gorhill** [reply]
Many EasyList lists do use uBO syntax[1], just not main EasyList/EasyPrivacy, and its fine this way otherwise we would have to mind conflicts between uBO's and EasyList lists.

---

[1] See `easylist_cookie/easylist_cookie_specific_uBO.txt` in EasyList Cookies, `fanboy-addon/fanboy_annoyance_specific_uBO.txt` in `EasyList – Annoyances`, etc

**Reply by MasterKia** [reply]
> See `easylist_cookie/easylist_cookie_specific_uBO.txt` in EasyList Cookies,

That's because Brave enables EasyList Cookies by default and they can't do with cookie notices using just the ABP syntax.

**Reply by gorhill** [reply]
Well uBO benefits from this works, all those filters is something we didn't have to take care here. In the end we all have the same burden of fixing countless filter issues, last thing I want is to criticize that other filter list maintainers are not doing enough, we are all in the same boat.

**Reply by peace2000** [reply]
@gorhill As we have pre-parsing directives for many browsers and popular adblockers, I wonder if Brave's own adblocker should have it's own directive also? What do you think?

https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#if-condition

**Reply by Yuki2718** [reply]
@peace2000 See https://github.com/brave/adblock-rust/issues/234. Haven't heard back from them.

**Reply by peace2000** [reply]
Sorry I was stupid, didn't realize that Brave adblock of course must support it as well.

**Comment by gorhill** [comment]
If there is an issue regarding what we are doing here, Brave people are welcome to discuss it.

If changes in uBO are causing issues in Brave or some uBO forks, you can be sure it's not intentional, I do what I think needs to be done, and many changes are things I wanted to do since a long time, it's not just done on a whim. Being free to work on the code without restrictions about breaking stuff out there is what allow uBO to evolve fast, so that would be difficult if I started to mind whether any change breaks some other projects in which I am not involved.

**Reply by MasterKia** [reply]
> some rules would be very lenghty and contain multiple commas, but is that an issue? I do that all the time

> The only issue I see here is pride

> because it literally takes 5 seconds to escape commas in arguments.

> But again, pride or not wanting to help others or whatever

If you want to cause change, this is not the way.

**Reply by Yuki2718** [reply]
We are all volunteers spending our personal time for others (literally hours a day and almost everyday), and you want to take more of our time by such a lengthy comment? Probably nobody other than @MasterKia will read. I don't remember your contirbution here.

**Comment by peace2000** [comment]
Sometimes cookie values sites create (especially json-based values) can be very lenghty and contain multiple commas. It's highly unpractical to escape every single one of them.

Also, if you want to compare a cookie value in the list to the value a site creates, comparison can't be done unless you first check and remove backslashes. This is unnecessary manual work.

I get the point of the OP but in this case I think it's the best to wait for the fix as it seems to already on it's way.

If Brave finds a rule invalid, why not just discard it? Why it crashes?
